src/components/Text.tsx

```diff
@@ -1,59 +1,52 @@
-import React from 'react';
-import PropTypes from 'prop-types';
-import _ from 'underscore';
+import React, {ForwardedRef} from 'react';
 // eslint-disable-next-line no-restricted-imports
-import {Text as RNText} from 'react-native';
+import {Text as RNText, StyleProp} from 'react-native';
+import type {TextStyle} from 'react-native';
 import fontFamily from '../styles/fontFamily';
 import themeColors from '../styles/themes/default';
 import variables from '../styles/variables';
 
-const propTypes = {
+type TextProps = {
     /** The color of the text */
-    color: PropTypes.string,
+    color?: string;
 
     /** The size of the text */
-    fontSize: PropTypes.number,
+    fontSize?: number;
 
     /** The alignment of the text */
-    textAlign: PropTypes.string,
+    textAlign?: 'left' | 'right' | 'auto' | 'center' | 'justify' | undefined;
```

Comment:
Do we need `undefined` if it's optional?
```suggestion
    textAlign?: 'left' | 'right' | 'auto' | 'center' | 'justify';
```